### PR TITLE
[eyes] Create eyes package on top of core

### PR DESCRIPTION
**Pull request by kyrylodolynskyi**
I am renaming `@applitools/eyes-api` to `@applitools/eyes` the new package depends on `@applitools/core` and provides user-facing classes based on the implementation in the core.

### CODE REVIEW

File: js/packages/eyes-api/test/unit/enums.spec.ts [hunk 1]
```diff
@@ -31,12 +31,12 @@ describe('enums', () => {
     })
   })
 
-  describe.skip('AndroidDeviceName', () => {
+  describe('AndroidDeviceName', () => {
```
Comment (zamboney) [inline js/packages/eyes-api/test/unit/enums.spec.ts:34]: 👍

**Comment by amitzur** [comment]
@kyrylodolynskyi while I'm reviewing this, a quick comment -
I love the simple name `@applitools/eyes` on one hand. But on the other hand I already see how it's going to be trouble communicating around it. The word `eyes` is so loaded with multiple meaning already. WDYT about it?
Other possible options I can think of - (a) keep the name `eyes-api` and just do a major version, (b) `eyes-sdk`, (c) `sdk-api`.

**Comment by kyrylodolynskyi** [comment]
@amitzur I don't like the `eyes-api`, because it is not just about an api anymore, it is a fully functional SDK. And traditionally all our SDK packages were called `eyes-*`, so it was only logical for me to call it `eyes` because it is everything.
Here are my pro-`eyes` arguments:
1. I would say that the `Eyes` class is the central entity in this package, so `eyes` is the perfect name.
2. This package going to be a user-facing package for integrators, so it should have a "cool" name - `eyes` is cool 😎.
3. It would make sense for integrators, they will see `eyes-selenium`, `eyes-webdriverio`, etc. and naturally they want `eyes-<their product>`, so they use the `eyes` package as a base for that.

**Comment by amitzur** [comment]
@kyrylodolynskyi fair enough, I agree with your reasoning. So when we refer to it we'll just say "the eyes package, you know the one that was eyes-api but now it's also core. Yeah, the eyes package" :)

**Review by amitzur — CHANGES_REQUESTED** [review]
I think this is great in simplifying things. The small inline comments I have are about the ability to understand the code for anyone who is new to it.
There are lots of variables like `sdk`, `core`, `eyes`, which have different meaning in different places. While TS helps, it's still hard to easily understand the code. I'd be happy if we adopted a convention that for example `Core.Eyes` variables would be called `coreEyes` or something like that. Same goes to `sdkOptions` like in my inline comment.

File: js/packages/eyes-api/src/SDK.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return {
+    core: getCore(options),
+    spec: options.spec,
+  }
+}
+
+const getCore = utils.general.cachify(function getCore<TSpec extends SpecType = SpecType>(
+  options: SDKOptions<TSpec> = {},
+): Core<TSpec, 'classic' | 'ufg'> {
+  return (options.makeCore ?? makeCore)({agentId: `js/eyes/${require('../package.json')}`, ...options})
```
[the remark this answers is not in the document]
Comment (kyrylodolynskyi) [reply js/packages/eyes-api/src/SDK.ts:25]: I want to keep a way to have a custom core implementation specified, for testing purposes. Also by default, we are using normal js core, but nothing should prevent us from using universal client instead.

File: js/packages/eyes-api/src/Eyes.ts [hunk 3]
```diff
@@ -201,7 +209,7 @@ export class Eyes<TSpec extends SpecType = SpecType> {
     viewportSizeOrSessionType?: RectangleSize | SessionType,
     sessionType?: SessionType,
   ): Promise<TSpec['driver'] | void> {
-    if (this._spec.isDriver?.(driverOrConfigOrAppName)) {
+    if (this._spec?.isDriver?.(driverOrConfigOrAppName)) {
```
[the remark this answers is not in the document]
Comment (kyrylodolynskyi) [reply js/packages/eyes-api/src/Eyes.ts:212]: In images

File: js/packages/eyes-api/src/Eyes.ts [hunk 4]
```diff
@@ -40,24 +40,28 @@ import {ValidationResult} from './output/ValidationResult'
...
 import {EyesRunner, ClassicRunner} from './Runners'
 import {Logger} from './Logger'
+import * as utils from '@applitools/utils'
 
-export class Eyes<TSpec extends SpecType = SpecType> {
-  protected static readonly _spec: CoreSpec
-  protected get _spec(): CoreSpec<TSpec> {
-    return (this.constructor as typeof Eyes)._spec as CoreSpec<TSpec>
+export class Eyes<TSpec extends Core.SpecType = Core.SpecType> {
+  protected static readonly _sdk: SDKOptions<Core.SpecType>
```
[the remark this answers is not in the document]
Comment (kyrylodolynskyi) [reply js/packages/eyes-api/src/Eyes.ts:46]: I will change it

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -376,7 +376,7 @@ jobs:
 
   # JS SDKS
   images:
-    needs: [ setup, core, api, test-utils ]
+    needs: [ setup, core, test-utils ]
```
[the remark this answers is not in the document]
Comment (kyrylodolynskyi) [reply .github/workflows/publish.yml]: fixed

File: js/packages/eyes-api/src/SDK.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyrylodolynskyi) [reply js/packages/eyes-api/src/SDK.ts]: I don't want to specify every property specifically, what I want is to pass `options` through and give agentId a default value. I made a change to make it more obvious and avoid passing `makeCore` to `makeCore`.

**Pull request merged**